From 7cf77b704544b6012105f7ed19807b798b6cb242 Mon Sep 17 00:00:00 2001
From: Yuan Zhao <yuanzhao@ti.com>
Date: Mon, 24 Jun 2019 13:24:11 -0500
Subject: [PATCH] Fix g++ 8.3.0 OpenCL example undefined behavior

- Fix undefined behavior of non-void functions with no return statements,
  when using g++ 8.3.0.  Change them to void functions.

Upstream-Status: Submitted [Remove this patch when next release is out]

Signed-off-by: Yuan Zhao <yuanzhao@ti.com>
---
 examples/persistent_common/host_assist.cpp | 2 +-
 examples/persistent_common/host_assist.h   | 2 +-
 examples/sgemm/main.cpp                    | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/examples/persistent_common/host_assist.cpp b/examples/persistent_common/host_assist.cpp
index f1640bc..fd8bb18 100644
--- a/examples/persistent_common/host_assist.cpp
+++ b/examples/persistent_common/host_assist.cpp
@@ -34,7 +34,7 @@ void print_completion_code(uint32_t completion_code)
 /******************************************************************************
 * Given an OpenCL Device, Abort if the device is not part of the AM57x platform
 ******************************************************************************/
-int assert_am57x(cl::Device &device)
+void assert_am57x(cl::Device &device)
 {
     cl::Platform platform;
     std::string  platform_name;
diff --git a/examples/persistent_common/host_assist.h b/examples/persistent_common/host_assist.h
index dec381e..4abf837 100644
--- a/examples/persistent_common/host_assist.h
+++ b/examples/persistent_common/host_assist.h
@@ -7,7 +7,7 @@
 /*-----------------------------------------------------------------------------
 * Prototypes
 *----------------------------------------------------------------------------*/
-int  assert_am57x         (cl::Device &device);
+void assert_am57x         (cl::Device &device);
 void print_completion_code(uint32_t completion_code);
 
 #endif // _HOST_ASSIST_H_
diff --git a/examples/sgemm/main.cpp b/examples/sgemm/main.cpp
index 85f19ac..dfac651 100644
--- a/examples/sgemm/main.cpp
+++ b/examples/sgemm/main.cpp
@@ -96,7 +96,7 @@ int NUMCOMPUNITS                    = 0;
 /* ======================================================================== */
 void PrintUsageAndExit();
 void HandleOptions(int argc, char* argv[]);
-bool SetSgemmParams(Device& device);
+void SetSgemmParams(Device& device);
 
 /* ======================================================================== */
 /*  MAIN                                                                    */
@@ -428,7 +428,7 @@ static cl_ulong roundDownPower2(cl_ulong value)
 /*-----------------------------------------------------------------------------
 * Check platform name, set sgemm blocking/tiling parameters accordingly
 *----------------------------------------------------------------------------*/
-bool SetSgemmParams(Device& device)
+void SetSgemmParams(Device& device)
 {
    int APanelSz        = 8  << 10;
    int BPanelSz        = 16 << 10;
-- 
2.17.1

